src/plugins/intel_cpu/tests/unit/transformations/convert_to_causalmask.cpp

```diff
@@ -0,0 +1,184 @@
+    auto copy__Broadcast = makeOP<ov::opset1::Broadcast>({masked_fill_Select, copy__ShapeOf, Constant_47319},
+                                                         {{"mode", "numpy"}});  //  tensor_array<f32[?,1,8192,..8192]>
+    auto SliceAssign_201_Reshape_2 =
+        makeOP<ov::opset1::Reshape>({copy__Broadcast, {-1}}, {{"special_zero", false}});  //  tensor_array<f32[?]>
+    auto SliceAssign_201_ScatterNDUpdate = makeOP<ov::opset4::ScatterNDUpdate>(
+        {SliceAssign_201_Reshape_0, SliceAssign_201_Reshape_1, SliceAssign_201_Reshape_2},
+        {});  //  tensor_array<f32[?]>
+    auto SliceAssign_201_Reshape_3 =
+        makeOP<ov::opset1::Reshape>({SliceAssign_201_ScatterNDUpdate, {-1, 1, max_seq_len, max_seq_len}},
+                                    {{"special_zero", true}});  //  tensor_array<f32[?,1,8192,8192]>
+    auto ScatterUpdate_93554 =
+        makeOP<ov::opset3::ScatterUpdate>({{0, 0, 0, 0}, {3}, kvLen, {0}}, {});  //  tensor_array<i32[4]>
+    auto slice_Slice_14 =
+        makeOP<ov::opset1::StridedSlice>({SliceAssign_201_Reshape_3, {0, 0, 0, 0}, ScatterUpdate_93554, {1, 1, 1, 1}},
+                                         {{"begin_mask", {1, 1, 1, 0}},
+                                          {"end_mask", {1, 1, 1, 0}},
+                                          {"new_axis_mask", {}},
+                                          {"shrink_axis_mask", {}},
+                                          {"ellipsis_mask", {}}});  //  tensor_array<f32[?,1,8192,..8192]>
+    auto index_Gather = makeOP<ov::opset8::Gather>({slice_Slice_14, cache_positions, 2},
+                                                   {{"batch_dims", 0}});  //  tensor_array<f32[?,1,?,..8192]>
+    auto result = index_Gather;
+
+    return std::make_shared<ov::Model>(ov::NodeVector{result},
+                                       ov::ParameterVector{attention_mask, batch_size, cache_positions, kvLen});
+}
+
+TEST_F(TransformationTestsF, ConvertToCausalMaskPreprocess) {
+    disable_rt_info_check();
+    disable_result_friendly_names_check();
+    const int batch = 2;
+    const int qLen = 16;
+    const int kvLen = 16;
+    const size_t max_seq_len = 2048;
+
+    model = buildCausalMaskPreprocess(batch, qLen, kvLen, max_seq_len);
+    manager.register_pass<VNodeFusion>();
+
+    {
+        auto attention_mask =
+            std::make_shared<ov::opset1::Parameter>(ov::element::i64, ov::Shape{batch, static_cast<size_t>(kvLen)});
+        auto batch_size = std::make_shared<ov::opset1::Parameter>(ov::element::i32, ov::Shape{1});
+        auto cache_positions =
+            std::make_shared<ov::opset1::Parameter>(ov::element::i32, ov::Shape{static_cast<size_t>(qLen)});
+        auto kvLen = std::make_shared<ov::opset1::Parameter>(ov::element::i32, ov::Shape{1});
+
+        auto result = makeOP<VNode>({attention_mask, batch_size, cache_positions, kvLen},
+                                    {{"config.type", "CausalMaskPreprocess"}});
+
+        model_ref =
```

Comment:
The test case here only constructs two graphs, but there is no comparison

Reply:
I confirm the comparison is done inside `TransformationTestsF::TearDown()`, and it would fail if the model didn't match with the pattern.

Reply:
That's good, I forget you use this TransformationTestsF template.

Reply:
But I just found it has no accuracy checks, I'm working on using subgraph_tests instead